### PR TITLE
Feature treat driver path as option

### PR DESCRIPTION
**Pull request by GreatBahram**
Hi there,

I just wanted to show it is possible to treat the driver path like this, which:
* now it is the responsibility of the argparse to check for the existence of DRIVER_PATH
*  (Maybe) it will be also easier to make this project work with other webdriver, in the future.

### CODE REVIEW

File: carbon/cli.py [hunk 1]
```diff
@@ -13,6 +13,13 @@ def get_code_from_file(filename: str) -> str:
         return fp.read()
 
 
+def environ_or_required(key: str) -> dict:
+    if os.environ.get(key):
+        return {"default": os.environ.get(key)}
```
Comment (bbelderbos) [inline carbon/cli.py:18]: nit pick, can do lookup once:

```
env_var = os.environ.get(key)
if env_var is None:
    return {"required": True}
else:
    return {"default": env_var}
```

File: carbon/cli.py [hunk 2]
```diff
@@ -55,4 +62,9 @@ def get_args():
         default=os.getcwd(),
         help="Specify folder where image should be stored (defaults to current directory)",
     )
+    parser.add_argument(
+        "--driver-path",
+        help="Path to the executable, if it is not given it reads value from environment variable (DRIVER_PATH)",
+        **environ_or_required("DRIVER_PATH"),
```
Comment (bbelderbos) [inline carbon/cli.py:68]: nice, so if set in environment this argument is optional, if not set it becomes required, right?

**Review by bbelderbos — APPROVED** [review]
Cool, small change if you don't mind

File: carbon/cli.py (hunk 1 above)
Comment (GreatBahram) [reply carbon/cli.py:18]: Nice suggestion!

File: carbon/cli.py (hunk 2 above)
Comment (GreatBahram) [reply carbon/cli.py:68]: Yes, that's exactly right!

Comment (GreatBahram) [reply carbon/cli.py:69]: By the way, click and other command-line applications provide this out-of-box, but we have to do this here.

Comment (bbelderbos) [reply carbon/cli.py:69]: Good to know :)
Thanks

**Comment by bbelderbos** [comment]
nice, thanks @GreatBahram

**Comment by bbelderbos** [comment]
@GreatBahram shall we start use the develop branch for this? Would need to set it up as default branch somewhere I believe ...

**Comment by bbelderbos** [comment]
FYI updating .env-template and README to use the renamed env variable `DRIVER_PATH`, also fixing this test that broke (thanks GH Actions): https://github.com/PyBites-Open-Source/pybites-carbon/runs/3660790462?check_suite_focus=true

**Comment by GreatBahram** [comment]
> @GreatBahram shall we start use the develop branch for this? Would need to set it up as default branch somewhere I believe ...

Sounds great to me, though it sometimes makes the workflow for a small project more complicated than it should be.
But, overall, it is good to have a separate branch for merging feature branches without messing with the release branch.

**Comment by GreatBahram** [comment]
> FYI updating .env-template and README to use the renamed env variable `DRIVER_PATH`, also fixing this test that broke (thanks GH Actions): https://github.com/PyBites-Open-Source/pybites-carbon/runs/3660790462?check_suite_focus=true

I wasn't aware of those places! Should I address these problems in this branch? or another one?

**Comment by bbelderbos** [comment]
I am happy to go with main only for now. No worries I will commit those fixes on main for now, bit busy with meetings and I want to try to automate the upload to PyPI when we add a new tag ...

**Pull request merged**